api/agent.go

```diff
@@ -0,0 +1,59 @@
+package api
+
+import (
+	"encoding/json"
+	"syscall"
+
+	ocispecs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type AgentConfig struct {
+	// ConnProto: the method to connect to agent(or proxy)
+	// "unix": the paths are unix socket
+	// "vsock": use cid and port pair to connect to the agent (kata-only)
+	// "kvmtool": the paths are kvmtool driver created pty slaver with escaping character
+	ConnProto string
+	// AgentProto: the agent protocol
+	// "hyperstart": orginal hyperstart protocol
+	// "kata": kata agent protocol
+	// "kata-yamux": kata agent protocol over yamux
+	AgentProto string
+
+	HyperstartCtlPath, HyperstartStreamPath string
+	KataPath                                string
+	KataVsockCid, KataVsockPort             uint32
```

Comment:
These also need documentation especially `KataPath` which is not a self-explaining name.